src/transactions/FeeBumpTransactionFrame.cpp

```diff
@@ -158,7 +161,26 @@ FeeBumpTransactionFrame::processPostApply(Application& app,
 {
     // We must forward the Fee-bump source so the refund is applied to the
     // correct account
-    mInnerTx->processPostApply(app, ltx, meta, getFeeSourceID());
+    int64_t refund = mInnerTx->processRefund(app, ltx, meta, getFeeSourceID());
```

Comment:
This might be a bit of a footgun if we add logic to `processPostApply` and/or if we decide to call `processPostApply` for the inner tx (the latter would cause double refund actually).
I think we should at least add a comment explaining these caveats. I don't see an obvious way to make this safer, maybe that requires the overhaul of the tx frame (there is that old issue about decoupling tx frame from tx result).

Reply:
Yeah that's why I initially went with the logic in https://github.com/stellar/stellar-core/pull/4228/commits/ad6dbbfa7734c6f71f9a66650fc4eeb10a331b9f, where I just apply the diff of the inner feeCharged pre vs post refund, but that's also not great. I'll add a comment.